src/kernel/sgd_momentum/cpu.cc

```diff
@@ -0,0 +1,96 @@
+/*! @copyright (c) 2022-present Skolkovo Institute of Science and Technology
+ *                              (Skoltech), Russia. All rights reserved.
+ *                 2023-present Artificial Intelligence Research Institute
+ *                              (AIRI), Russia. All rights reserved.
+ *
+ * NNTile is a software framework for fast training of big neural networks on
+ * distributed-memory heterogeneous systems based on the StarPU runtime system.
+ *
+ * @file src/kernel/sgd_momentum/cpu.cc
+ * Fused SGD with momentum step on buffers on CPU
+ *
+ * @version 1.0.0
+ * */
+
+#include "cpu.hh"
+#include <cmath>
+#include "nntile/kernel/cpu.hh"
+
+namespace nntile::kernel::sgd_momentum
+{
+
+template<typename T>
+void cpu(Index num_iter, Index num_elems, Scalar momentum_, Scalar lr_,
+         Scalar weight_decay_, const T *grad, T *velocity, T *p)
```

Comment:
Formulas inside are incorrect and do not take as a parameter "dampening" factor and "nesterov".